build-system/release-workflows/cut-nightly.js

```diff
@@ -48,48 +39,79 @@ async function cutNightlyBranch() {
   );
 
   for (const {sha} of commits.data) {
-    const commitStatus = await octokit.rest.repos.getCombinedStatusForRef({
-      ...params,
-      ref: sha,
-    });
-    const {state} = commitStatus.data;
-
-    log('Status of commit', cyan(sha), 'is', colorizeState(state));
-
-    if (state === 'success') {
-      const response = await octokit.rest.git.updateRef({
+    const {'check_runs': checkRuns} = (
+      await octokit.rest.checks.listForRef({
         ...params,
-        ref: 'heads/nightly',
-        sha,
-      });
+        ref: sha,
+      })
+    ).data;
+    if (checkRuns.some(({status}) => status != 'completed')) {
+      log(
+        'Not all check runs for commit',
+        cyan(sha),
+        'are completed. Checking next commit...'
+      );
+      continue;
+    }
 
-      // Casting to Number because the return type in Octokit is incorrectly
-      // annotated to only ever return 200.
-      switch (Number(response.status)) {
-        case 201:
-          log('Cut a new', cyan('nightly'), 'with', cyan(sha));
-          break;
-        case 200:
-          log(
-            'The',
-            cyan('nightly'),
-            'branch is already at the latest',
-            green('green'),
-            'sha',
-            cyan(sha)
-          );
-          break;
-        default:
-          log(
-            red('An error occurred while attempting to fast-forward the'),
-            cyan('nightly'),
-            red('branch to sha'),
-            cyan(sha)
-          );
-      }
+    if (
+      !checkRuns.every(({conclusion}) =>
+        ['success', 'neutral', 'skipped'].includes(conclusion ?? '')
```

Comment:
Why `??` here?

Reply:
`conclusion` is `string | null`

Reply:
But why coalesce? `[...].includes(null)` is still falsey

Reply:
For some reason I got a type error for that! I didn't dig in to it, but I assume it's because `['success', 'neutral', 'skipped']` is typed as `string[]`, and the `includes` method is probably of type `(searchElement: T) => boolean`, so `T === string` here

Reply:
Okay I dug in and my assumption was correct: https://github.com/microsoft/TypeScript/blob/main/lib/lib.es2016.array.include.d.ts#L27

Reply:
And someone wants to fix this: https://github.com/microsoft/TypeScript/pull/45767

Reply:
🕳️🐇

Reply:
Ah right, I've hit this before. I think I cast the array instead ¯\_(ツ)_/¯ carry on!